core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -94,7 +107,7 @@ object Queue {
       * @param queue    Queue, expressed as vector for fast cons/uncons from head/tail
       * @param deq      A list of waiting dequeuers, added to when queue is empty
       */
-    case class State(queue: Vector[A], deq: Vector[Async.Ref[F,A]])
+    case class State(queue: Vector[A], deq: Vector[Either[Async.Ref[F,A], Async.Ref[F,Vector[A]]]])
```

Comment:
`Vector` is indeed not specialized for singletons, or emptiness, or really anything that it would make *sense* for it to be specialized on.  `Chunk` would be a better choice.